File: packages/entity-graph-tauri/rust-plugin/android/src/main/java/EntityGraphPlugin.kt
package com.prometheusags.entitygraph

import android.app.Activity
import app.tauri.annotation.Command
import app.tauri.annotation.TauriPlugin
import app.tauri.plugin.Invoke
import app.tauri.plugin.JSObject
import app.tauri.plugin.Plugin

@TauriPlugin
class EntityGraphPlugin(activity: Activity) : Plugin(activity) {
    @Command
    fun ping(invoke: Invoke) {
        val response = JSObject()
        response.put("plugin", "entity-graph-tauri")
        response.put("platform", "android")
        invoke.resolve(response)
    }
}

File: packages/entity-graph-tauri/rust-plugin/ios/Sources/EntityGraphPlugin.swift
import Tauri
import WebKit

class EntityGraphPlugin: Plugin {
  @objc public func ping(_ invoke: Invoke) {
    invoke.resolve([
      "plugin": "entity-graph-tauri",
      "platform": "ios",
    ])
  }
}

@_cdecl("init_plugin_entity_graph_tauri")
func initPlugin() -> Plugin {
  return EntityGraphPlugin()
}
